universidade-estadual-paulista-faculdade-de-engenharia-de-ilha-solteira-abnt.csl

```diff
@@ -0,0 +1,989 @@
+        <if type="broadcast motion_picture song" match="any">
+          <group>
+            <text variable="title" text-case="uppercase" suffix=", "/>
+            <text macro="issued-year"/>
+          </group>
+        </if>
+        <else-if type="musical_score" match="any">
+          <group>
+            <text macro="author-short" suffix=", "/>
+            <text macro="issued-year"/>
+            <text macro="citation-locator" prefix=", "/>
+          </group>
+        </else-if>
+        <else>
+          <group>
+            <text macro="author-short" suffix=", "/>
+            <text macro="issued-year"/>
+            <text macro="citation-locator" prefix=", "/>
+          </group>
+        </else>
+      </choose>
+    </layout>
+  </citation>
+  <bibliography line-spacing="1" entry-spacing="1">
+    <sort>
+      <key macro="author-short"/>
+      <key variable="year-suffix"/>
+      <key macro="title"/>
+      <key variable="issued"/>
+    </sort>
+    <layout>
+      <choose>
+        <if type="article">
+          <group>
+            <text macro="author" suffix=". "/>
+            <text variable="authority" suffix=". "/>
+            <text macro="title" suffix=" "/>
+            <text variable="medium" suffix=". "/>
+            <text variable="number" prefix="Versão " suffix=". "/>
+            <text macro="publisher"/>
+            <text variable="genre" suffix="."/>
+            <text macro="access"/>
+          </group>
+        </if>
+        <else-if type="article-journal article-magazine" match="any">
+          <group>
+            <text macro="author" suffix=". "/>
+            <text variable="authority" suffix=". "/>
+            <text macro="title"/>
+            <text variable="genre" suffix=". "/>
+            <text macro="container-title" suffix=", "/>
+            <text macro="place"/>
+            <text macro="edition" suffix=", "/>
+            <text macro="locators" suffix=", "/>
+            <text macro="issued"/>
+            <text macro="access"/>
+          </group>
+        </else-if>
+        <else-if type="article-newspaper">
+          <group>
+            <text macro="author" suffix=". "/>
+            <text variable="authority" suffix=". "/>
+            <text macro="title"/>
+            <text macro="container-title" suffix=", "/>
+            <text macro="place" prefix=" "/>
+            <text macro="locators"/>
+            <text macro="access"/>
+          </group>
+        </else-if>
+        <else-if type="bill">
+          <choose>
+            <if variable="container-title" match="any">
+              <group>
+                <group delimiter=". " suffix=". ">
+                  <text macro="author"/>
+                  <text variable="authority"/>
+                  <text macro="title"/>
+                  <text variable="abstract"/>
+                </group>
+                <text macro="container-title" suffix=", "/>
+                <text macro="section" prefix="Seção " suffix=", "/>
```

Comment:
please use labels for all of these in the next few lines, don't harcode this into the affixes